### PR TITLE
feat: allow post to task with many tasks

### PR DESCRIPTION
**Pull request by joncodo**
# Summary

Users need to be able to query many tasks at once with the api

IE:

```
curl --location --request POST 'localhost:8080/task' \
--header 'Content-Type: application/json' \
--data-raw '[
    {
        "Plugin": "gitlab",
        "Options": {
            "projectId": 20103385
        }
    },
    {
        "Plugin": "jira",
        "Options": {
            "boardId": 8
        }
    }
]'
```

NOTE: This means you will have to send down an array of json objects. Otherwise, you get this error:

Screen Shot 2021-09-02 at 10 51 07 AM

### Key Points

- [x] This is a breaking change
- [x] New or existing documentation is updated

Screen Shot 2021-09-02 at 10 55 11 AM

### CODE REVIEW

File: api/task/task.go [hunk 1]
```diff
@@ -5,20 +5,31 @@ import (
...
+
+	tasks := make([]models.Task, len(data))
+
+	for _, value := range data {
+		task, err := services.CreateTask(value)
+		if err != nil {
+			_ = ctx.AbortWithError(http.StatusInternalServerError, err)
+			return
+		}
+		tasks = append(tasks, *task)
```
Comment (narrowizard) [inline api/task/task.go:31]: You should not use `append` with a certain length array, here the length of tasks will be double length as expacted.  

> https://play.golang.org/p/Aqk4KbigBZE

File: api/task/task.go [hunk 2]
```diff
@@ -5,20 +5,31 @@ import (
...
 
 func Post(ctx *gin.Context) {
-	var data services.NewTask
+	var data []services.NewTask
+
 	err := ctx.MustBindWith(&data, binding.JSON)
 	if err != nil {
-		_ = ctx.AbortWithError(http.StatusBadRequest, err)
+		logger.Error("", err)
+		ctx.JSON(http.StatusBadRequest, "You must send down an array of objects")
```
Comment (narrowizard) [inline api/task/task.go:19]: Why use `ctx.JSON` instead of `ctx.AbortWithError` ?

**Review by narrowizard — CHANGES_REQUESTED** [review]

File: api/task/task.go (hunk 1 above)
Comment (likyh) [reply api/task/task.go:31]: Unavailable For Legal Reasons
Viewing and/or sharing code snippets is not available in your country for legal reasons. This message might also appear if your country is misdetected. If you believe this is an error, please file an issue.

Comment (narrowizard) [reply api/task/task.go:31]: What about this? https://goplay.space/#Aqk4KbigBZE

File: api/task/task.go (hunk 2 above)
Comment (joncodo) [reply api/task/task.go:19]: I wanted to send back a message to the user

File: api/task/task.go (hunk 1 above)
Comment (joncodo) [reply api/task/task.go:31]: Seems like I need this: https://play.golang.org/p/4JjXhCgMvJL

Comment (joncodo) [reply api/task/task.go:31]: Screen Shot 2021-09-03 at 9 00 05 AM

**Pull request merged**